### PR TITLE
Add new reource about sls log config

### PR DESCRIPTION
**Pull request by xuxiaoahang2018**
hi
this pr:
Increase logconfg resource and related mono-tests and documents;

Increase logconfg to machine group resource and related mono-tests and documents;

### CODE REVIEW

File: alicloud/resource_alicloud_logtail_config.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+		Read:   resourceAlicoudLogtailConfigRead,
+		Update: resourceAlicoudLogtaiConfiglUpdate,
+		Delete: resourceAlicoudLogtailConfigDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+
+			"config_name": &schema.Schema{
+				Type:     schema.TypeString,
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_logtail_config.go]: I think just `name` is ok. Please modify `config_name` to `name`.

File: alicloud/resource_alicloud_logtail_config.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+				Required: true,
+				ForceNew: true,
+			},
+
+			"input_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_logtail_config.go]: Does the `input_type` have optional values?

File: alicloud/resource_alicloud_logtail_config.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+				ForceNew: true,
+			},
+			"logstore": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"output_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_logtail_config.go]: Does the `output_type` have optional values?

File: alicloud/resource_alicloud_logtail_config.go [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+		},
+	}
+}
+
+func resourceAlicoudLogtaiConfiglCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	var inputConfigInputDetail = make(map[string]interface{})
+	data := d.Get("input_detail").(string)
+	if err := json.Unmarshal([]byte(data), &inputConfigInputDetail); err != nil {
+		return err
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_logtail_config.go]: please use BuildWrapError to format the err.

File: alicloud/resource_alicloud_logtail_config.go [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+				InputDetail: logconfig.InputDetail,
+				OutputDetail: sls.OutputDetail{
+					ProjectName:  d.Get("project").(string),
+					LogStoreName: d.Get("logstore").(string),
+				},
+			})
+		})
+		if err != nil {
+			return fmt.Errorf("UpdateLogTailConfig %s got an error: %#v.", split[2], err)
+		}
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_logtail_config.go]: If there is only one api invoking, please remove the partial setting.

File: alicloud/resource_alicloud_logtail_config_test.go [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+		Steps: []resource.TestStep{
+			{
+				Config: testAlicloudLogTailbasic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAlicloudLogProjectExists("alicloud_log_project.example", &project),
+					testAccCheckAlicloudLogStoreExists("alicloud_log_store.example", &store),
+					testAccCheckAlicloudLogTailConfigExists("alicloud_logtail_config.example", &config),
+					resource.TestCheckResourceAttr("alicloud_logtail_config.example", "input_type", "file"),
+					resource.TestCheckResourceAttr("alicloud_logtail_config.example", "log_sample", "test"),
+					resource.TestCheckResourceAttr("alicloud_logtail_config.example", "config_name", "evan-terraform-config"),
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_logtail_config_test.go]: Please check all of attributes.

File: website/docs/r/logtail_config.html.markdown [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+}
+```
+## Argument Reference
+
+The following arguments are supported:
+
+* `project` - (Required, ForceNew) The project name to the log store belongs.
+* `logstore` - (Required, ForceNew) The log store name to the query index belongs.
+* `input_type` - (Required) The input type. Currently, only file is supported.
+* `log_sample` - The log sample of the Logtail configuration. The log size cannot exceed 1,000 bytes.
```
Comment (xiaozhu36) [inline website/docs/r/logtail_config.html.markdown]: If it is not required, please note it to optional.

File: website/docs/r/logtail_config.html.markdown [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+The following attributes are exported:
+
+* `id` - The ID of the log store index. It formats of `<project>:<logstore>:<config_name>`.
+* `project` - The project name.
+* `logstore` - Log store name.
+* `config_name` - The Logtail configuration name.
+* `log_sample` - The log sample of the Logtail configuration.
+* `output_type` - The output type.
+* `input_type` - The input type
+* `input_detail` - The logtail detailed input configuration.
```
Comment (xiaozhu36) [inline website/docs/r/logtail_config.html.markdown]: Just retain the `id` is ok in the output parameters.

File: alicloud/resource_alicloud_logtail_config.go (hunk 3 above)
Comment (xuxiaoahang2018) [reply alicloud/resource_alicloud_logtail_config.go]: Currently not optional, optional fields have been added to facilitate future changes

**Pull request closed without merging**